lib/glific/providers/gupshup/partner/partner_api.ex

```diff
@@ -0,0 +1,148 @@
+defmodule Glific.Providers.Gupshup.PartnerAPI do
+  @moduledoc """
+  A module to handle fetching tier related information like quality rating and app rating using partner API
+  """
+
+  alias Glific.Caches
+  alias Plug.Conn.Query
+
+  use Tesla
+  plug(Tesla.Middleware.Logger)
+
+  plug(Tesla.Middleware.FormUrlencoded,
+    encode: &Query.encode/1
+  )
+
+  @partner_url "https://partner.gupshup.io/partner/account"
+  @app_url "https://partner.gupshup.io/partner/app/"
+  @global_organization_id 0
+
+  # fetches partner token
+  @spec get_partner_token :: {:ok, map()} | {:error, any}
+  defp get_partner_token do
+    email = Application.fetch_env!(:glific, :gupshup_partner_email)
```

Comment:
{email, password}  =  Saas.partner-creadentials()

Reply:
SaaS table is not that flexible